Vic2ToHoI4/Source/V2World/Ai/Vic2AI.cpp

```diff
@@ -0,0 +1,49 @@
+#include "Vic2AI.h"
+#include "../World.h"
+#include "../Provinces/Province.h"
+#include "AIStrategy.h"
+#include "ParserHelpers.h"
+
+
+
+Vic2::Vic2AI::Vic2AI(std::istream& theStream)
+{
+	registerRegex("threat|antagonize|befriend|protect|rival",
+		 [this](const std::string& strategyType, std::istream& theStream) {
+			 const auto& strategy = AIStrategy(strategyType, theStream);
+			 aiStrategies.push_back(strategy);
+		 });
+	registerKeyword("conquer_prov", [this](const std::string& strategyType, std::istream& theStream) {
+		const auto& conquerStrategy = AIStrategy(strategyType, theStream);
+		conquerStrategies.push_back(conquerStrategy);
+	});
+	registerRegex(commonItems::catchallRegex, commonItems::ignoreItem);
+
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void Vic2::Vic2AI::consolidateConquerStrategies(const std::map<int, std::shared_ptr<Province>>& provinces)
+{
+	for (const auto& strategy: conquerStrategies)
+	{
+		if (const auto& theProvince = provinces.find(strategy.getProvID())->second; theProvince)
```

Comment:
Add a check that theProvince != provinces.end(). Stranger things have happened.

Reply:
![codacy production](https://user-images.githubusercontent.com/59363714/100806250-699e7900-3430-11eb-9d7a-4f2f6cebc4d5.PNG)
If it matters anything, theProvince is a shared_ptr?

Reply:
Ah, I missed the .second there. You'll need to actually split this into bits:
```
if (const auto& provinceItr = provinces.find(strategy.getProvID()); provinceItr != provinces.end())
{
   const auto& province = provinceItr.second;
   if (!province)
   {
      continue;
   }

   const auto& owner = theProvince->getOwner();
   ...
```